spec/src/main/asciidoc/servlet-spec-body.adoc

```diff
@@ -1303,6 +1303,169 @@ the header value to an `int`, a `NumberFormatException` is thrown. If
+  * Any `"."` or `".."` segment with any encoded characters (e.g. `/path/%2e%2e/info`)
+  * Any `".."` segment preceded by an empty segment (e.g. `/path//../info`)
+  * Any empty segment other than the last segment, with parameters (e.g. `/path/;param/info` )
+  * The `"\"` character encoded or not. (e.g. `/path\info`)
+  * Any control characters either encoded or not.  (e.g. `/path%00/info`)
+  * Any illegal hex sequences following a % character
+  * Any illegal UTF-8 code sequences.
+
+==== Example URIs
+> TODO more work needed on this section. Perhaps some parts need to be moved to other sections.
+
+Consider the following URIs with a forbidden security constraint at `/secret/*` and a servlet registered at /dispatch/* that forwards request to the string return from pathInfo()
+
+. Example URIs
+|===
+| Encoded URI path  | Decoded Path | Rejected
+
+| `foo/bar` | `/foo/bar` | 400 must start with /
+| `/foo/bar` | `/foo/bar` |
+| `/foo/bar;jsessionid=1234` | `/foo/bar` |
+| `/foo/bar/` | `/foo/bar/` |
+| `/foo/bar/;jsessionid=1234` | `/foo/bar/` |
+| `/foo;/bar;` | `/foo/bar` |
+| `/foo;/bar;/;` | `/foo/bar/` |
+| `/foo%00/bar/` | `/foo/bar/` | 400 control character
+| `/foo%7Fbar` | `/foobar` | 400 control character
+| `/foo%2Fbar` | `/foo/bar` | 400 encoded /
+| `/foo\bar` | `/foo\bar` | 400 backslash character
+| `/foo%5Cbar` | `/foo\bar` | 400 backslash character
+| `/foo;%2F/bar` | `/foo/bar` | 400 encoded /
+| `/foo/./bar` | `/foo/bar` |
+| `/foo/././bar` | `/foo/bar` |
+| `/./foo/bar` | `/foo/bar` |
+| `/foo/%2e/bar` | `/foo/bar` | 400 encoded dot segment
+| `/foo/.;/bar` | `/foo/bar` | 400 dot segment with parameter
+| `/foo/%2e;/bar` | `/foo/bar` | 400 encoded dot segment
+| `/foo/.%2Fbar` | `/foo/./bar` | 400 encoded /
+| `/foo/.%5Cbar` | `/foo/.\bar` | 400 backslash character
+| `/foo/bar/.` | `/foo/bar` |
+| `/foo/bar/./` | `/foo/bar/` |
+| `/foo/bar/.;` | `/foo/bar` | 400 dot segment with parameter
+| `/foo/bar/./;` | `/foo/bar/` |
+| `/foo/.bar` | `/foo/.bar` |
+| `/foo/../bar` | `/bar` |
+| `/foo/../../bar` | `/../bar` | 400 leading dot-dot-segment
+| `/../foo/bar` | `/../foo/bar` | 400 leading dot-dot-segment
+| `/foo/%2e%2E/bar` | `/bar` | 400 encoded dot segment
+| `/foo/%2e%2e/%2E%2E/bar` | `/../bar` | 400 leading dot-dot-segment & encoded dot segment
+| `/foo/./../bar` | `/bar` |
+| `/foo/..;/bar` | `/bar` | 400 dot segment with parameter
+| `/foo/%2e%2E;/bar` | `/bar` | 400 encoded dot segment
+| `/foo/..%2Fbar` | `/foo/../bar` | 400 encoded /
+| `/foo/..%5Cbar` | `/foo/..\bar` | 400 backslash character
+| `/foo/bar/..` | `/foo` |
+| `/foo/bar/../` | `/foo/` |
+| `/foo/bar/..;` | `/foo` | 400 dot segment with parameter
+| `/foo/bar/../;` | `/foo/` |
+| `/foo/..bar` | `/foo/..bar` |
+| `/foo/.../bar` | `/foo/.../bar` |
+| `/foo//bar` | `/foo/bar` |
+| `//foo//bar//` | `/foo/bar/` |
+| `/;/foo;/;/bar/;/;` | `/foo/bar/` | 400 empty segment with parameters
+| `/foo//../bar` | `/bar` |
```

Comment:
This has been described above as one of the suspicious sequences which should be rejected with 400 response.

Reply:
yeah... but that was really only if containers don't remove empty segments.   I think that either needs to be made clear or that rejection removed.